android/clientlib-rxjava/src/main/java/com/solana/mobilewalletadapter/clientlib/RxMobileWalletAdapter.java

```diff
@@ -0,0 +1,117 @@
+package com.solana.mobilewalletadapter.clientlib;
+
+import android.net.Uri;
+
+import androidx.annotation.CheckResult;
+import androidx.annotation.IntRange;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.Size;
+
+import com.solana.mobilewalletadapter.clientlib.protocol.MobileWalletAdapterClient.AuthorizeResult;
+import com.solana.mobilewalletadapter.clientlib.protocol.MobileWalletAdapterClient.ReauthorizeResult;
+import com.solana.mobilewalletadapter.clientlib.protocol.MobileWalletAdapterClient.SignAndSendTransactionResult;
+import com.solana.mobilewalletadapter.clientlib.protocol.MobileWalletAdapterClient.SignPayloadResult;
+import com.solana.mobilewalletadapter.clientlib.protocol.RxMobileWalletAdapterClient;
+import com.solana.mobilewalletadapter.clientlib.scenario.RxLocalAssociationScenario;
+import com.solana.mobilewalletadapter.common.protocol.CommitmentLevel;
+
+import java.util.function.Function;
+
+import io.reactivex.Completable;
+import io.reactivex.Single;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+
+public class RxMobileWalletAdapter {
+
+    @NonNull
+    private final RxLocalAssociationScenario mRxLocalAssociationScenario;
+
+    public RxMobileWalletAdapter(@IntRange(from = 0) int clientTimeoutMs) {
+        this.mRxLocalAssociationScenario = new RxLocalAssociationScenario(clientTimeoutMs);
+    }
+
+    @CheckResult
+    @NonNull
+    public Single<AuthorizeResult> authorize(@Nullable Uri identityUri,
+                                             @Nullable Uri iconUri,
+                                             @Nullable String identityName) {
+        return startExecuteAndClose(
```

Comment:
Same comment as left on the Kotlin PR: https://github.com/solana-mobile/mobile-wallet-adapter/pull/101#discussion_r919466107

Reply:
Yeah let's have a chat about it!